courses/models.py

```diff
@@ -286,6 +300,24 @@ def first_unexpired_run(self):
             lambda course_run: course_run.is_unexpired,
         )
 
+    @property
+    def unexpired_run_dates(self):
+        """
+        Gets all the start dates for unexpired CourseRuns associated with this Course
+
+        Returns:
+           [Datetime]: list of start_dates from all unexpired course runs
+        """
+        return list(
+            map(
+                lambda unexpired_course_run: unexpired_course_run.start_date,
+                filter(
+                    lambda course_run: course_run.is_unexpired,
+                    self.courseruns.all().order_by("start_date"),
```

Comment:
suggestion:

can we break it simply ? I mean simply filter runs, build and return a data structure for it
```
course_runs = self.courseruns.filter(is_unexpired=True).order_by('start_date')
.... 
```

Reply:
`is_unexpired` is a compound check of several attributes and a bit of logic that can't simply be moved to SQL.